libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -236,15 +266,15 @@ private static bool CompositeContainsEntity(EntityRecommendation compositeEntity
             return entity.StartIndex >= compositeEntityMetadata.StartIndex &&
                    entity.EndIndex <= compositeEntityMetadata.EndIndex;
         }
-        
+
         /// <summary>
         /// If a property doesn't exist add it to a new array, otherwise append it to the existing array
         /// </summary>
         private static void AddProperty(JObject obj, string key, JToken value)
         {
             if (((IDictionary<string, JToken>)obj).ContainsKey(key))
```

Comment:
This if statement makes my head hurt. :( 

I cannot realistically expect to debug this. IS there a way to decompose so it's clearer to read and/or more debuggable in the future?

Reply:
Leaving as is.